cachito/workers/pkg_managers/pip.py

```diff
@@ -1514,3 +1517,101 @@ def upload_raw_package(repo_name, artifact_path, dest_dir, filename, is_request_
     to_nexus_hoster = not is_request_repository
     log.debug("Uploading %r as a raw package to the %r Nexus repository", artifact_path, repo_name)
     nexus.upload_raw_component(repo_name, dest_dir, components, to_nexus_hoster)
+
+
+def generate_pip_conf_content(nexus_repo_identifier, username, password, custom_ca_path=None):
+    """
+    Generate a pip.conf file with the nexus repo url and authentication configured.
+
+    :param str nexus_repo_identifier: the nexus repo identifier,
+        e.g. cachito-pip-hosted-<request_id>
+    :param str username: the username of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str password: the password of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str custom_ca_path: the path to set ``cert`` to in the pip.conf file; if not provided,
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
+    # https://help.sonatype.com/repomanager3/formats/npm-registry#npmRegistry-AuthenticationUsingBasicAuth
+    # Create the request specific nexus repo url to be passed as `index-url` within pip.conf
+    nexus_repo_url = base64.b64encode(
+        f"https://[{username}[:{password}]@]{nexus_repo_identifier}".encode("utf-8")
+    ).decode("utf-8")
+
+    pip_conf_content = textwrap.dedent(
+        f"""\
+        [global]
+        index-url = {nexus_repo_url}
+        """
+    )
+    if custom_ca_path:
+        # The CA could be embedded in the actual pip.conf file with the `cert` option, but this
+        # would mean that the CA file contents would be duplicated in the Cachito database for
+        # every request that uses a pip conf file
+        pip_conf_content += f'cert = "{custom_ca_path}"\n'
+
+    return pip_conf_content
+
+
+def create_custom_pip_conf_file(request_id, package_configs=None):
+    """
+    Create a custom pip.conf file.
+
+    It should configure pip to point to the request specific repository in Nexus.
+
+    :param int request_id: the ID of the Cachito request
+    :param list package_configs: the list of optional package configurations submitted by
+    the user
+    :rtype: str
+    :return: The file path for pip configuration file that points to the request
+    specific repository in Nexus
```

Comment:
This seems to return a list of dicts rather than a string

Reply:
Yes, I was only returning the single pip file earlier but then later changed the logic and forgot to update the doc strings. Thanks for catching this.